FirebaseSwiftUI/FirebaseAuthSwiftUI/Sources/Services/AuthService.swift

```diff
@@ -169,19 +169,19 @@ public final class AuthService {
 
   public func signIn(credentials credentials: AuthCredential) async throws {
     authenticationState = .authenticating
-    if currentUser?.isAnonymous == true, configuration.shouldAutoUpgradeAnonymousUsers {
-      try await linkAccounts(credentials: credentials)
-    } else {
-      do {
+    do {
+      if currentUser?.isAnonymous == true, configuration.shouldAutoUpgradeAnonymousUsers {
```

Comment:
Question - If the user is signed-in anonymously,  and the user subsequently signs in with email credential, it will raise the exception `email-already-in-use`. If we get this exception, do we just sign the user in with email credential without linking?

Reply:
Also - there should be an opportunity to merge backend data if linking fails, some form of callback to be run before signing user in with credential. This code snippet demonstrates what I mean (I couldn't find equivalent iOS sample): https://github.com/firebase/snippets-web/blob/467eaa165dcbd9b3ab15711e76fa52237ba37f8b/snippets/auth-next/link-multiple-accounts/auth_merge_accounts.js#L8

Reply:
You can copy the merge behavior in the ObjC FirebaseUI implementation here: https://github.com/firebase/FirebaseUI-iOS/blob/main/FirebaseEmailAuthUI/Sources/FUIEmailAuth.m#L397